nifi-nar-bundles/nifi-jmx-bundle/nifi-jmx-processors/src/main/java/org/apache/nifi/processors/jmx/GetJMX.java

```diff
@@ -0,0 +1,406 @@
+        whiteListFilter = new ListFilter( "" );
+        blackListFilter = new ListFilter( "" );
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        queue.clear();
+    }
+
+    private Set<Record> performQuery( ProcessContext context) throws ProcessException {
+        PropertyValue hostname = context.getProperty(HOSTNAME);
+        PropertyValue port = context.getProperty(PORT);
+        PropertyValue whitelist = context.getProperty(WHITELIST);
+        PropertyValue blacklist = context.getProperty(BLACKLIST);
+
+        Set<Record> metricsSet = new HashSet<Record>();
+
+        whiteListFilter.setListString(whitelist.getValue());
+        blackListFilter.setListString( blacklist.getValue() );
+
+        try {
+            HashMap<String,Object> env = new HashMap<String,Object>();
+
+            StringBuilder urlStr = new StringBuilder("service:jmx:rmi:///jndi/rmi://");
+            urlStr.append(hostname).append(":");
+            urlStr.append(port);
+            urlStr.append("/jmxrmi");
+
+            JMXServiceURL url = new JMXServiceURL(urlStr.toString());
+            JMXConnector jmxc = JMXConnectorFactory.connect(url, null);
+
+            MBeanServerConnection mbsc = jmxc.getMBeanServerConnection();
+
+            Set<ObjectName> mbeans = mbsc.queryNames(null, null);
+            for (ObjectName mbean : mbeans) {
+                String domain = mbean.getDomain();
+
+                Record record = new Record();
+                Hashtable<String, String> mbeanProperties = mbean.getKeyPropertyList();
+
+                String type = (mbeanProperties.containsKey("type")) ? mbeanProperties.get("type") : "";
+
+                if (canProcess(domain, type)) {
+                    record.setDomain( domain );
+                    record.setProperties(mbeanProperties.entrySet());
+
+                    MBeanInfo mbeanInfo = mbsc.getMBeanInfo(mbean);
+                    MBeanAttributeInfo[] mbeanAttributeInfoAr = mbeanInfo.getAttributes();
+
+                    boolean validRecord = true;
+
+                    Map<String, String> mbeanStringMap = new HashMap<String, String>();
+
+                    for (MBeanAttributeInfo mbeanAttributeInfo : mbeanAttributeInfoAr) {
+                        try {
+                            Object value = mbsc.getAttribute(mbean, mbeanAttributeInfo.getName());
+
+                            mbeanStringMap.put( mbeanAttributeInfo.getName(), value.toString() );
+                        } catch (Exception e) {
+                            // IF ERROR DO NOT ADD to metricsSet
+                            validRecord = false;
+                            logger.warn("Exception fetching MBean attribute: " + e.getMessage() +
```

Comment:
please add the exception as argument of the log message (same comment each time you're logging a message when catching an exception, it's really helpful to have more than just the message)